plugins/builds/update.js

```diff
@@ -359,7 +361,7 @@ module.exports = () => ({
 
                 // Start stage teardown build if stage is done
                 if (stageTeardownBuild && stageTeardownBuild.status === 'CREATED') {
-                    const stageIsDone = await isStageDone({ stage, event: newEvent });
+                    const stageIsDone = await isStageDone({ stage, event: newEvent, buildStatus: newBuild.status });
```

Comment:
Should we use stage build status instead of the job build status?
```suggestion
                    const stageIsDone = await isStageDone({ stage, event: newEvent, stageBuildStatus: stageBuild.status });
```